docs/repositories.md

```diff
@@ -21,24 +22,282 @@ on PyPI.
+
+{{% /note %}}
+
+#### Default Package Source
+
+By default, Poetry configures [PyPI](https://pypi.org) as the default package source for your
+project. You can alter this behaviour and exclusively look up packages only from the configured
+package sources by adding a **single** source with `default = true`.
+
+```bash
+poetry source add --default foo https://foo.bar/simple/
+```
+
+{{% warning %}}
+
+Configuring a custom package source as default, will effectively disable [PyPI](https://pypi.org)
+as a package source for your project.
 
-Adding a new repository is easy with the `config` command.
+{{% /warning %}}
+
+#### Secondary Package Sources
+
+If package sources are configured as secondary, all it means is that these will be given a lower
+priority when selecting compatible package distribution that also exists in your default package
+source.
+
+You can configure a package source as a secondary source with `secondary = true` in your package
+source configuration.
 
 ```bash
-poetry config repositories.foo https://foo.bar/simple/
+poetry source add --secondary foo https://foo.bar/simple/
 ```
 
-This will set the url for repository `foo` to `https://foo.bar/simple/`.
+There can be more than one secondary package source.
+
+{{% note %}}
+
+Package sources, even when configured as secondary, will be searched when Poetry looks up packages
+despite the package having been found in the default source or another configured source.
 
-### Configuring credentials
+If you wish to avoid this, you may explicitly specify which source too look in for a particular
+package.
+
+```bash
+poetry add --source pypi httpx
+```
+
+{{% /note %}}
+
+### Supported Package Sources
+
+#### Python Package Index (PyPI)
+
+Poetry interacts with [PyPI](https://pypi.org) via its
+[JSON API](https://warehouse.pypa.io/api-reference/json.html). This is used to retrieve a requested
+package's versions, metadata, files etc.
+
+{{% note %}}
+
+If the the package's published metadata is invalid, Poetry will download available bdist/sdist to
+inspect it locally to identify the relevant metadata.
+
+{{% /note %}}
+
+If you want to explicitly select a package from [PyPI](https://pypi.org) you can use the `--source`
+option with the [`add`]({{< relref "cli#add" >}}) command, like shown below.
+
+```bash
+poetry add --source pypi httpx@^0.22.0
+```
+
+This will generate the following configuration snippet in your `pyproject.toml` file.
+
+```toml
+httpx = {version = "^0.22.0", source = "pypi"}
+```
+
+{{% warning %}}
+
+If any source within a project is configured with `default = true`, The implicit `pypi` source will
+be disabled and not used for any packages.
+
+{{% /warning %}}
+
+#### Simple API Repository
+
+Poetry can fetch and install package dependencies from public or private custom repositories that
+implement the simple API repository as described in [PEP 503](https://peps.python.org/pep-0503/).
```

Comment:
```suggestion
implement the simple repository API as described in [PEP 503](https://peps.python.org/pep-0503/).
```

Reply:
This one is as intended as it is "simple API" repository.

Reply:
I do think that's a lot less clear, and it doesn't match the title of the PEP. At the very least "simple API repository specification" would make it more clear.